docs/messaging/messaging-spans.md

```diff
@@ -195,21 +197,64 @@ Examples:
 * `AuthenticationRequest-Conversations process`
 * `(anonymous) publish` (`(anonymous)` being a stable identifier for an unnamed destination)
 
-### Span kind
-
-A producer of a message should set the span kind to `PRODUCER` unless it synchronously waits for a response: then it should use `CLIENT`.
-The processor of the message should set the kind to `CONSUMER`, unless it always sends back a reply that is directed to the producer of the message
-(as opposed to e.g., a queue on which the producer happens to listen): then it should use `SERVER`.
-
 ### Operation names
 
 The following operations related to messages are defined for these semantic conventions:
 
 | Operation name | Description |
 | -------------- | ----------- |
-| `publish`      | A message is sent to a destination by a message producer/client.       |
-| `receive`      | A message is received from a destination by a message consumer/server. |
-| `process`      | A message that was previously received from a destination is processed by a message consumer/server. |
+| `publish`      | One or more messages are provided for publishing to an intermediary. |
+| `create`       | A message is created. |
+| `receive`      | One or more messages are requested by a consumer. |
+| `deliver`      | One or more messages are passed to a consumer. |
+
+### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if no `create` spans are present. |
```

Comment:
What should the Span kind be if create spans are present?

Reply:
Proposal from today's SIG meeting:
> `PRODUCER`, if the context of the publish span is used as creation context.

And add this elaboration from the OTEP:
> For cases not covered by the table above, the span kind should be set according to the [generic specification about span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind), e. g. it should be set to CLIENT if the publish spans models a synchronous call to the intermediary.
>
> Setting span kinds according to this table ensures that span links between consumers and producers always exist between a PRODUCER span on the producer side and a CONSUMER span on the consumer side. This allows analysis tools to interpret linked traces without the need for additional semantic hints.